### PR TITLE
Structured data improvement

### PR DESCRIPTION
**Pull request by opportus**
Implementing dynamically generated json for categories or archives is somehow a little bit delicate...
Instead of remaking a loop to get each post data, I believe I better use an existing loop such as `storefront_loop_post`... The little problem with this approche is that I need to store the generated data from the loop outside of the loop for later use... So for this, I use a Storefront static property.
As far as my actual competences are concerned, I don't see better approche.
Let me know if you'd like me to modify anything anyway...

### CODE REVIEW

File: inc/woocommerce/storefront-woocommerce-template-functions.php [hunk 1]
```diff
@@ -315,5 +315,47 @@ function storefront_handheld_footer_bar_cart_link() {
...
+  }
+}
+
+if ( ! function_exists( 'storefront_woocommerce_init_structured_data' ) ) {
+  /**
+   * Generate product category structured data...
+   * Hooked into the `woocommerce_before_shop_loop_item` action...
+   * Apply the `storefront_woocommerce_structured_data` filter hook for structured data customization...
+   */
+  function storefornt_woocommerce_init_structured_data() {
```
Comment (jameskoster) [inline inc/woocommerce/storefront-woocommerce-template-functions.php]: Typo - 'storefornt' xD

File: inc/woocommerce/storefront-woocommerce-template-hooks.php [hunk 2]
```diff
@@ -64,6 +64,13 @@
 add_action( 'storefront_header', 'storefront_product_search', 40 );
 add_action( 'storefront_header', 'storefront_header_cart',    60 );
 
+/**
+ * Structured Data
+ *
+ * @see storefront_woocommerce_init_structured_data()
+ */
+add_action( 'woocommerce_before_shop_loop_item', 'storefornt_woocommerce_init_structured_data' );
```
Comment (opportus) [inline inc/woocommerce/storefront-woocommerce-template-hooks.php]: I was wondering how it managed to work with your pointed typo error.
Surprisingly, I made the same here without copy pasting the function name !

Comment (jameskoster) [inline inc/woocommerce/storefront-woocommerce-template-hooks.php]: Hah! A convenient typo

**Comment by jameskoster** [comment]
Cheers, I'll check / merge this soon.

**Pull request merged**